sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/serializer/JsonSerializer.java

```diff
@@ -16,22 +16,22 @@
      * Reads a JSON stream into its object representation.
      *
      * @param stream JSON stream.
-     * @param clazz {@link Class} representing the object.
+     * @param typeReference {@link TypeReference} representing the object.
      * @param <T> Type of the object.
      * @return The object represented by the deserialized JSON stream.
      */
     @Override
-    <T> Mono<T> deserialize(InputStream stream, Class<T> clazz);
+    <T> Mono<T> deserialize(InputStream stream, TypeReference<T> typeReference);
```

Comment:
This serves the purpose. The only thing is that the user will now have to learn about `TypeReference` but since this is going to be used mainly client library developers, I am not too worried about this.